src/xpi.js

```diff
@@ -0,0 +1,80 @@
+import yauzl from 'yauzl';
+
+
+export default class Xpi {
+
+  constructor(filename, zipLib=yauzl) {
+    this.filename = filename;
+    this.zipLib = zipLib;
+    this.metadata = null;
```

Comment:
Why initialise to `null`? We can test for truthiness with `undefined`.

Is it to establish the class props in the code? That's cool, but just curious. I like `this.foo = undefined;` or even `this.foo;` if it's just about expressing intent/prop names via code. `null` is different from `undefined`, but I'm being semantic.